packages/ia-user-list/src/ia-user-list.ts

```diff
@@ -0,0 +1,190 @@
+   */
+  @query('#list-id') private listId: HTMLInputElement;
+
+  /**
+   * @private
+   * @type {HTMLInputElement}
+   * @memberof IAUserList
+   */
+  @query('#list-name') private listName: HTMLInputElement;
+
+  /**
+   * @private
+   * @type {HTMLInputElement}
+   * @memberof IAUserList
+   */
+  @query('#list-description') private listDescription: HTMLInputElement;
+
+  /**
+   * @private
+   * @type {HTMLInputElement}
+   * @memberof IAUserList
+   */
+  @query('#list-private') private listPrivate: HTMLInputElement;
+
+  private async saveListDetails(event: Event) {
+    event.preventDefault();
+
+    try {
+      const requestInit: RequestInit = {};
+      requestInit.credentials = 'include';
+      requestInit.method = 'PUT';
+      requestInit.body = JSON.stringify({
+        listId: this.listId.value,
+        listName: this.listName.value,
+        listDescription: this.listDescription.value,
+        listPrivate: this.listPrivate.checked,
+      });
+
+      const response = await fetch(this.apiURL, requestInit);
+      console.log('response', response);
+
+      this.dispatchEvent(
+        new CustomEvent('listDetailsSaved', {
+          detail: {
+            inputData: requestInit,
+            outputData: response,
+          },
+        })
+      );
+    } catch (error) {
+      console.log('error', error);
+    }
+  }
+
+  private emitCloseModalEvent(event: Event) {
+    event.preventDefault();
+    event.stopPropagation();
+    this.dispatchEvent(new Event('listModalClosed'));
+  }
+
+  render() {
+    return html`
+      <section class="new-list">
+        <form @submit=${this.saveListDetails}>
+          <div class="field">
+            <input type="hidden" id="list-id" .value=${this.listData?.listId} />
+            <label for="list-name">List name*</label>
+            <input
+              type="text"
+              id="list-name"
+              placeholder="Silver age comics are the best"
+              value=${this.listData?.listName}
+              required
+            />
+          </div>
+          <div class="field">
+            <label for="list-description">Description</label>
+            <textarea
+              id="list-description"
+              placeholder="Great comics from the silver age of 1970"
+            >
+${this.listData?.description}</textarea
+            >
+          </div>
+          <div class="field">
+            <label for="list-private">Private list</label>
+            <input
+              type="checkbox"
+              id="list-private"
+              checked="${this.listData?.private}"
+            />
+          </div>
+          <div class="footer field">
+            <button
+              class="ia-button dark"
+              id="cancel"
+              @click=${(event: Event) => {
+                this.emitCloseModalEvent(event);
+              }}
+            >
+              Cancel
+            </button>
+            <button class="ia-button primary">Save</button>
```

Comment:
noting callback is needed here.

Reply:
Yes, I already use a callback function when you save something